### PR TITLE
[9.x] Command for dump current config values

### PR DESCRIPTION
**Pull request by Davidnadejdin**
#40821

### CODE REVIEW

File: src/Illuminate/Foundation/Console/ConfigDumpCommand.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    $key,
+                    is_array($value) ? '' : $value,
+                ];
+            });
+
+            $this->table(
+                ['Key', 'Value'],
+                $rows,
+            );
+        } elseif (is_string($rawConfig)) {
```
Comment (rezaamini-ir) [inline src/Illuminate/Foundation/Console/ConfigDumpCommand.php:48]: I guess it's better to refactor in this way:

```php
if(is_string($rawConfig)){
    $this->info($rawConfig);
    return 0;
}

// Handle if it is an array
```

It's much more cleaner.

Comment (Davidnadejdin) [reply src/Illuminate/Foundation/Console/ConfigDumpCommand.php:48]: but then we will duplicate the same return. This is a too small command

**Comment by driesvints** [comment]
I think we shouldn't do this. Dumping config values onto the CLI leaves you to the risk of exposing sensitive info in logs like the APP_KEY env variable.

**Comment by Davidnadejdin** [comment]
> I think we shouldn't do this. Dumping config values onto the CLI leaves you to the risk of exposing sensitive info in logs like the APP_KEY env variable.

command output can get into the log?

File: src/Illuminate/Foundation/Console/ConfigDumpCommand.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+use Illuminate\Console\Command;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+
+class ConfigDumpCommand extends Command
+{
+    /**
+     * @var string
+     */
+    protected $signature = 'config:dump {id?}';
```
Comment (lkmadushan) [inline src/Illuminate/Foundation/Console/ConfigDumpCommand.php:15]: maybe we should use `key` is here?

Comment (Davidnadejdin) [reply src/Illuminate/Foundation/Console/ConfigDumpCommand.php:15]: Maybe. But i focused on it:
https://github.com/php-fig/container/blob/master/src/ContainerInterface.php#L22

**Comment by axlon** [comment]
My 2 cents: I don't think this adds a lot of value since Laravel already ships with tinker and its not very difficult or cubersome to access your config through tinker when debugging.

**Comment by Davidnadejdin** [comment]
> My 2 cents: I don't think this adds a lot of value since Laravel already ships with tinker and its not very difficult or cubersome to access your config through tinker when debugging.

you can also get a list of routes
```
Illuminate\Support\Facades\Route::getRoutes();
```

config:dump is small helper command, not a big feature 🙂

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**